File: objectModel/TypeScript/Persistence/CdmFolder/Projections/OperationAddSupportingAttributePersistence.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import {
    CdmCorpusContext,
    cdmObjectType,
    CdmOperationAddSupportingAttribute,
    CdmTypeAttributeDefinition,
    copyOptions,
    resolveOptions
} from '../../../internal';
import { TypeAttributePersistence } from '../TypeAttributePersistence';
import { OperationAddSupportingAttribute } from '../types';
import * as utils from '../utils';
import { OperationBasePersistence } from './OperationBasePersistence';

/**
 * Operation AddSupportingAttribute persistence
 */
export class OperationAddSupportingAttributePersistence {
    public static fromData(ctx: CdmCorpusContext, object: OperationAddSupportingAttribute): CdmOperationAddSupportingAttribute {
        if (!object) {
            return undefined;
        }

        const addSupportingAttributeOp: CdmOperationAddSupportingAttribute = OperationBasePersistence.fromData(ctx, cdmObjectType.operationAddSupportingAttributeDef, object);
        addSupportingAttributeOp.supportingAttribute = utils.createAttribute(ctx, object.supportingAttribute) as CdmTypeAttributeDefinition;

        return addSupportingAttributeOp;
    }

    public static toData(instance: CdmOperationAddSupportingAttribute, resOpt: resolveOptions, options: copyOptions): OperationAddSupportingAttribute {
        if (!instance) {
            return undefined;
        }

        const data: OperationAddSupportingAttribute = OperationBasePersistence.toData(instance, resOpt, options);
        data.supportingAttribute = TypeAttributePersistence.toData(instance.supportingAttribute, resOpt, options)

        return data;
    }
}

File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/projections/OperationAddSupportingAttributePersistence.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.projections;

import com.fasterxml.jackson.databind.JsonNode;
import com.microsoft.commondatamodel.objectmodel.cdm.CdmCorpusContext;
import com.microsoft.commondatamodel.objectmodel.cdm.CdmTypeAttributeDefinition;
import com.microsoft.commondatamodel.objectmodel.cdm.projections.CdmOperationAddSupportingAttribute;
import com.microsoft.commondatamodel.objectmodel.enums.CdmObjectType;
import com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.Utils;
import com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.types.projections.OperationAddSupportingAttribute;
import com.microsoft.commondatamodel.objectmodel.utilities.CopyOptions;
import com.microsoft.commondatamodel.objectmodel.utilities.ResolveOptions;

/**
 * Operation AddSupportingAttribute persistence
 */
public class OperationAddSupportingAttributePersistence {
    public static CdmOperationAddSupportingAttribute fromData(final CdmCorpusContext ctx, final JsonNode obj) {
        if (obj == null) {
            return null;
        }

        CdmOperationAddSupportingAttribute addSupportingAttributeOp = OperationBasePersistence.fromData(ctx, CdmObjectType.OperationAddSupportingAttributeDef, obj);

        if (obj.get("supportingAttribute") != null) {
            addSupportingAttributeOp.setSupportingAttribute((CdmTypeAttributeDefinition) Utils.createAttribute(ctx, obj.get("supportingAttribute")));
        }

        return addSupportingAttributeOp;
    }

    public static OperationAddSupportingAttribute toData(final CdmOperationAddSupportingAttribute instance, final ResolveOptions resOpt, final CopyOptions options) {
        if (instance == null) {
            return null;
        }

        OperationAddSupportingAttribute obj = OperationBasePersistence.toData(instance, resOpt, options);
        obj.setSupportingAttribute(Utils.jsonForm(instance.getSupportingAttribute(), resOpt, options));

        return obj;
    }
}
